lib/mozart_fetcher/components/ares.ex

```diff
@@ -0,0 +1,7 @@
+defmodule MozartFetcher.Components.Ares do
+  alias MozartFetcher.Components.Ares
+  @moduledoc false
+  @derive [Jason.Encoder]
+  defstruct [:index, :id, :status, :data]
+  @type t :: %Ares{index: non_neg_integer(), id: String.t(), status: non_neg_integer(), data: Map.t()}
```

Comment:
```suggestion
  @type t :: %__MODULE__{index: non_neg_integer(), id: String.t(), status: non_neg_integer(), data: Map.t()}
```
I _think_ you can do that, then there is no need for the alias on line 2.